File: referralhero/src/main/java/com/sdk/rh/RHUtil.java
package com.sdk.rh;

import android.content.Context;
import android.content.pm.ApplicationInfo;
import android.content.pm.PackageManager;
import android.content.res.Resources;


/**
 * Class for RH utility methods
 */

public class RHUtil {



    /**
     *Read RH accessToken from Android mainfests meta-data tag
     */

    public static String readRhKey(Context context) {
        String RHKey = null;

        String metaDataKey = "com.rh.sdk.RhKey";
        // manifest overrides string resources
        try {
            final ApplicationInfo ai = context.getPackageManager().getApplicationInfo(context.getPackageName(), PackageManager.GET_META_DATA);
            if (ai.metaData != null) {
                RHKey = ai.metaData.getString(metaDataKey);
                if (RHKey == null) {
                    RHKey = ai.metaData.getString("com.rh.sdk.RhKey");
                }
            }
        } catch (final PackageManager.NameNotFoundException ignore) { }
        if (RHKey != null) return RHKey;

        // check string resources as the last resort
        Resources resources = context.getResources();
        RHKey = resources.getString(resources.getIdentifier(metaDataKey, "string", context.getPackageName()));

        return RHKey;
    }

    /**
     *Read RH Campaign ID  from Android mainfests meta-data tag
     */

    public static String readRhCampaignID(Context context) {
        String RHKey = null;

        String metaDataKey = "com.rh.sdk.uuid";
        // manifest overrides string resources
        try {
            final ApplicationInfo ai = context.getPackageManager().getApplicationInfo(context.getPackageName(), PackageManager.GET_META_DATA);
            if (ai.metaData != null) {
                RHKey = ai.metaData.getString(metaDataKey);
                if (RHKey == null) {
                    RHKey = ai.metaData.getString("com.rh.sdk.uuid");
                }
            }
        } catch (final PackageManager.NameNotFoundException ignore) { }
        if (RHKey != null) return RHKey;

        // check string resources as the last resort
        Resources resources = context.getResources();
        RHKey = resources.getString(resources.getIdentifier(metaDataKey, "string", context.getPackageName()));

        return RHKey;
    }

}

File: referralhero/src/main/java/com/sdk/rh/RHUtil.kt
package com.sdk.rh

import android.content.Context
import android.content.pm.PackageManager
import java.util.*

/**
 * Class for RH utility methods
 */
object RHUtil {
    /**
     * Read RH accessToken from Android mainfests meta-data tag
     */
    fun readRhKey(context: Context): String {
        var RHKey: String? = null
        val metaDataKey = "com.rh.sdk.ApiKey"
        // manifest overrides string resources
        try {
            val ai = context.packageManager.getApplicationInfo(
                context.packageName,
                PackageManager.GET_META_DATA
            )
            if (ai.metaData != null) {
                RHKey = ai.metaData.getString(metaDataKey)
                if (RHKey == null) {
                    RHKey = ai.metaData.getString("com.rh.sdk.ApiKey")
                }
            }
        } catch (ignore: PackageManager.NameNotFoundException) {
        }
        if (RHKey != null) return RHKey

        // check string resources as the last resort
        val resources = context.resources
        RHKey =
            resources.getString(resources.getIdentifier(metaDataKey, "string", context.packageName))
        return RHKey
    }

    /**
     * Read RH Campaign ID  from Android mainfests meta-data tag
     */
    fun readRhCampaignID(context: Context): String {
        var RHKey: String? = null
        val metaDataKey = "com.rh.sdk.uuid"
        // manifest overrides string resources
        try {
            val ai = context.packageManager.getApplicationInfo(
                context.packageName,
                PackageManager.GET_META_DATA
            )
            if (ai.metaData != null) {
                RHKey = ai.metaData.getString(metaDataKey)
                if (RHKey == null) {
                    RHKey = ai.metaData.getString("com.rh.sdk.uuid")
                }
            }
        } catch (ignore: PackageManager.NameNotFoundException) {
        }
        if (RHKey != null) return RHKey

        // check string resources as the last resort
        val resources = context.resources
        RHKey =
            resources.getString(resources.getIdentifier(metaDataKey, "string", context.packageName))
        return RHKey
    }

    fun formatString(format: String?, vararg args: Any?): String? {
        return java.lang.String.format(Locale.US, format, args)
    }

}